src/stirling/source_connectors/socket_tracer/bcc_bpf/protocol_inference.h

```diff
@@ -60,6 +60,46 @@ static __inline enum message_type_t infer_http_message(const char* buf, size_t c
   return kUnknown;
 }
 
+static __inline enum message_type_t infer_tls_message(const char* buf, size_t count) {
```

Comment:
Have you run the confusion matrix analysis on this? @chengruizhe had done some work there. 

You're looking at multiple bytes, so I'm hoping itll be okay', but still wondering if it conflicts with detection of other protocols.

Reply:
I have not since I was assuming until #2095 is implemented that this would be always be disabled by default. I'm pretty familiar with the confusion matrix since I ported running it to bazel. I'll add this logic there and see the resulting matrix.

Reply:
Unfortunately New Relic wasn't able to share the entire data set used to generate the [current matrix](https://github.com/pixie-io/pixie/blob/main/src/stirling/protocol_inference/bi_directional_connection_matrix.png), so the matrices below are based on their acceptable subset with packet captures I collected to fill in the gaps.

Before these tls changes:
![matrix_before_tls](https://github.com/user-attachments/assets/a7d69013-9f4c-43a9-8ced-0d8177434a89)

After these tls changes:
![matrix_after_tls](https://github.com/user-attachments/assets/915deb4e-d041-473f-841e-dabe7ae372c8)

[Here](https://github.com/pixie-io/pixie/commit/720a6e0f5ba1fd3dd07c30ebedb84766207433b4) were the code changes to generate the "after tls matrix"